notifiers/slack/notifier.go

```diff
@@ -18,66 +15,46 @@ type payload struct {
 
 // ArgsSlackNotifier defines the arguments needed to create a new slack notifier
 type ArgsSlackNotifier struct {
-	Config *config.Slack
+	Config     *config.Slack
+	HTTPClient notifiers.HTTPClient
 }
 
 type slackNotifier struct {
-	config *config.Slack
+	url        string
+	httpClient notifiers.HTTPClient
 }
 
 // NewSlackNotifier will create a new email notifier instance
-func NewSlackNotifier(args ArgsSlackNotifier) (*slackNotifier, error) {
+func NewSlackNotifier(args *ArgsSlackNotifier) (*slackNotifier, error) {
```

Comment:
could have kept the argument DTO passed as value (same reason: a client struct can not alter the same argument used - maybe - on another client struct)

Reply:
changed to pass the argument as value